internal/entities/handlers/strategies/entity/add_originating_entity.go

```diff
@@ -135,16 +149,21 @@ func (_ *addOriginatingEntityStrategy) GetName() string {
 func (_ *addOriginatingEntityStrategy) upsertLegacyEntity(
 	ctx context.Context,
 	entType minderv1.Entity,
-	parentEwp *models.EntityWithProperties, childProps *properties.Properties,
+	parentEwp *models.EntityWithProperties, pbEnt protoreflect.ProtoMessage,
 	t db.ExtendQuerier,
 ) (uuid.UUID, error) {
 	var legacyId uuid.UUID
 
 	switch entType { // nolint:exhaustive
 	case minderv1.Entity_ENTITY_PULL_REQUESTS:
+		pr, ok := pbEnt.(*minderv1.PullRequest)
+		if !ok {
+			return uuid.Nil, fmt.Errorf("unexpected proto message type: %T", pbEnt)
+		}
+
 		dbPr, err := t.UpsertPullRequest(ctx, db.UpsertPullRequestParams{
 			RepositoryID: parentEwp.Entity.ID,
-			PrNumber:     childProps.GetProperty(ghprop.PullPropertyNumber).GetInt64(),
+			PrNumber:     pr.Number,
```

Comment:
this is much better than leaking the ghprops module but we should eventually either get rid of the number in the PR protobuf message or even better get rid of the protobuf message completely.

Reply:
We'll only get here once we get rid of the PR table.